### PR TITLE
Add missing `invitable` field

### PR DESCRIPTION
**Pull request by MrPowerGamerBR**
The `invitable` field controls this option

image

### CODE REVIEW

**Comment by HopeBaron** [comment]
shouldn't this be added to the create data as well

**Comment by MrPowerGamerBR** [comment]
> shouldn't this be added to the create data as well

Yes, I forgot about that, let me add it there too

**Comment by MrPowerGamerBR** [comment]
Done 😊

**Comment by HopeBaron** [comment]
I guess you missed the thread create builder as well

**Comment by MrPowerGamerBR** [comment]
> I guess you missed the thread create builder as well

I think that a ThreadCreateBuilder doesn't exist (yet) :(

image

**Comment by HopeBaron** [comment]
it's called the StartThreadRequest but since properties are being added to it maybe make a builder if you will.

**Comment by MrPowerGamerBR** [comment]
I added the `StartThreadBuilder`, however I'm not sure if the changes that I made to the core module are correct

File: core/src/main/kotlin/behavior/channel/TextChannelBehavior.kt [hunk 1]
```diff
@@ -69,18 +72,24 @@ interface TextChannelBehavior : PrivateThreadParentChannelBehavior {
...
+        reason: String? = null,
     ): TextChannelThread {
-        return unsafeStartThread(name, archiveDuration, ChannelType.PublicGuildThread, reason) as TextChannelThread
+        return unsafeStartThread(
+            name,
+            archiveDuration,
+            ChannelType.PublicGuildThread
+        ) {
+            this.reason = reason
+        } as TextChannelThread
```
Comment (HopeBaron) [inline core/src/main/kotlin/behavior/channel/TextChannelBehavior.kt:83]: remove the reason from here, add the builder as a part of the function
so: 
```kt 
suspend fun startPublicThread(
        name: String,
        archiveDuration: ArchiveDuration = ArchiveDuration.Day,
        builder: StartThread.() -> Unit
    ): TextChannelThread
```
and then pass it to unsafe

File: rest/src/main/kotlin/builder/channel/thread/StartThreadBuilder.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import dev.kord.common.entity.ArchiveDuration
+import dev.kord.common.entity.ChannelType
+import dev.kord.common.entity.optional.OptionalBoolean
+import dev.kord.common.entity.optional.delegate.delegate
+import dev.kord.common.entity.optional.optional
+import dev.kord.rest.builder.AuditRequestBuilder
+import dev.kord.rest.json.request.StartThreadRequest
+
+class StartThreadBuilder(var name: String, var autoArchiveDuration: ArchiveDuration, var type: ChannelType) : AuditRequestBuilder<StartThreadRequest> {
```
Comment (HopeBaron) [inline rest/src/main/kotlin/builder/channel/thread/StartThreadBuilder.kt:11]: make type a `val` here, you don't want to make a switch in the types which makes the builder no longer useful.

**Review by HopeBaron — CHANGES_REQUESTED** [review]

File: rest/src/main/kotlin/builder/channel/thread/StartThreadBuilder.kt (hunk 2 above)
Comment (MrPowerGamerBR) [reply rest/src/main/kotlin/builder/channel/thread/StartThreadBuilder.kt:11]: Only `val` for the `ChannelType`, or for `name` and `autoArchiveDuration` too?

Comment (HopeBaron) [reply rest/src/main/kotlin/builder/channel/thread/StartThreadBuilder.kt:11]: `type` only

**Pull request merged**